sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BatchLoads.java

```diff
@@ -77,12 +91,23 @@
     this.singletonTable = singletonTable;
     this.dynamicDestinations = dynamicDestinations;
     this.destinationCoder = destinationCoder;
+    this.maxNumWriters = DEFAULT_MAX_NUM_WRITERS;
   }
 
   void setTestServices(BigQueryServices bigQueryServices) {
     this.bigQueryServices = bigQueryServices;
   }
 
+  /** Get the maximum number of file writers that will be open simultaneously in a bundle. */
+  public int getMaxNumWriters() {
+    return maxNumWriters;
+  }
+
+  /** Set the maximum number of file writers that will be open simultaneously in a bundle. */
+  public void setMaxNumWriters(int maxNumWriters) {
```

Comment:
The multi-core worker policy is Dataflow specific, but added a caveat in the comment above (by DEFAULT_MAX_NUM_WRITERS_PER_BUNDLE.

Runners are free to override this parameter based on extra information the runner knows (how much memory will be on the workers, how many bundles will run in parallel, etc.).